### ISSUE TITLE
Dark or black blocks in inventory after long playing

### ISSUE THREAD

**Issue by Wuzzy2** [minetest/minetest#4876]
I occassionally stumple upon a very annoying graphical glitch which appears rarely and I haven't figured out what triggers it.

Sometimes, after playing for a while without exiting Minetest, some of the blocks in the inventory may appear to be totally black or just very darkish.

I don't know what triggers this mod, but I have a few observations of which I am not really 100% sure of:

* If I recall correctly, the blocks turn black after joining a game, but not directly in the 
* It seems the bug is somehow related to Lua errors. E.g. the bug appears to happen if the current Minetest session went through a couple of Lua errors
* The graphical glitch goes away after you restart Minetest

I experience this bug in 0.4.14 on GNU/Linux.
Sadly, I have no screenshot at hand or instructions on how to reproduce it.

**Comment by bigfoot547** [comment]
:+1:
This has happened to me.

**Comment by DS-Minetest** [comment]
i had dark nodes in inventory on my old pc all the time
if i find a screenshot with this bug i will show it here

**Comment by Fixer-007** [comment]
You are using OpenGL render I hope?

**Comment by DS-Minetest** [comment]
as much as i know, there was no option for renderer

**Comment by Wuzzy2** [comment]
Yes, I was also using the OpenGL renderer when this happened. I don't use the other renderers.

**Comment by Wuzzy2** [comment]
I just got this bug in 0.4.14:
![Black nodes](https://i.imgur.com/RQVaMmm.png)

Some observations:
- Full “classic” cubes ARE affected
- Leaf nodes ARE affected
- Nodeboxes are NOT affected
- Luminant nodes (e.g. mese lamp) are NOT affected
- Nodes like glass are NOT affected (idk why. My guess is that this is either because of `drawtype="glasslike"` or because of `paramtype="light"` or both)
- All items which are not nodes are NOT affected

I still have no idea why I got this. :-(

**Comment by Wuzzy2** [comment]
kilbith just found that bug, too:

![Black crafting guide](https://lut.im/E2vMJaklqE/CeDyuRt6qYtH4nVU.png)

kilbith had this in 2886f0c (but a dirty version, kilbith had some own code added :-/ so take this with a grain of salt )

It looks like the bug still happens in dev versions.

**Comment by Fixer-007** [comment]
Wuzzy, can you bisect it?

**Comment by Wuzzy2** [comment]
I try that the next time it happens to me. Please note the black nodes always go away after restarting Minetest.

**Comment by sofar** [comment]
```
2016-12-12 22:05:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
```

this is fixed in minetest_game master, please check you're not using 0.4.14 or an old snapshot.

**Comment by sofar** [comment]
This is a lighting issue. I wonder if it has anything to do with the recent light code changes. @lhofhansl or @juhdanad may want to take a look?

I can't reproduce this, at least on NV drivers/linux.

**Comment by Wuzzy2** [comment]
> I wonder if it has anything to do with the recent light code changes.

Probably not. I found this in 0.4.14 (stable) as well.

> I can't reproduce this, at least on NV drivers/linux.

Well, this bug is pretty sneaky. It occours when I least expect it (but at least I know it *may* occour after a new game join). Which makes it hard to track down. :-(

**Comment by juhdanad** [comment]
Could you convert one node's texture (for example desert sand) to a new sRGB profile?
Here's how to do: https://wiki.archlinux.org/index.php/Libpng_errors

If next time you get this bug, and desert sand is not affected anymore, then we found the fix.

**Comment by sfan5** [comment]
@sofar 

>```
>2016-12-12 22:05:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
>```
>
>this is fixed in minetest_game master, please check you're not using 0.4.14 or an old snapshot.

Incorrect color profiles only cause slight changes in color and this log message only _warns_ about them, it **doesn't** say that some image could not be read.

Besides the bug would either always happen or never happen because images are not reloaded at runtime.

**Comment by Wuzzy2** [comment]
@sfan5: Sorry about the sRGB confusion. It seems the sRGB warning appears *regardless* off the black node item images.

> EDIT: The sRGB profile warnings also appear when the nodes are not black. So this may be just a false alert.

**Comment by beyondlimits** [comment]
Could Minetest go out of texture space? Look at the first screenshot. There's not only nodes in the hud dark, there's missing texture of the landscape too.

**Comment by DS-Minetest** [comment]
@beyondlimits 
If you mean my screenshot: Some textures of the dirt are not there because there was an unloaded mapblock, I think.
But as much as I remember I also had problems with texture space: When I had loaded too many mods with textures, some nodes had not the right textures, it were textures of other nodes. (As much as I remember.)

**Comment by bigfoot547** [comment]
@sfan5 I think that this bug is confirmed. Right?

**Comment by paramat** [comment]
@Wuzzy2 does this still happen?

**Comment by Wuzzy2** [comment]
Hmmm, I don't know. I don't play Minetest that often anymore. Especially not bleeding edge. The problem is, the bug is rare.
But I cannot recall having it seen in 0.4.17.1.
